app/controllers/api/v1/directory_listings_controller.rb

```diff
@@ -0,0 +1,322 @@
+module Api
+  module V1
+    class DirectoryListingsController < ApiBaseController
+      include Swagger::Blocks
+
+      before_action :set_study
+      before_action :check_study_permission
+      before_action :set_directory_listing, except: [:index, :create]
+
+      respond_to :json
+
+      swagger_path '/studies/{study_id}/directory_listings' do
+        operation :get do
+          key :tags, [
+              'DirectoryListings'
+          ]
+          key :summary, 'Find all DirectoryListings in a Study'
+          key :description, 'Returns all DirectoryListings for the given Study'
+          key :operationId, 'study_directory_listings_path'
+          parameter do
+            key :name, :study_id
+            key :in, :path
+            key :description, 'ID of Study'
+            key :required, true
+            key :type, :string
+          end
+          response 200 do
+            key :description, 'Array of DirectoryListing objects'
+            schema do
+              key :type, :array
+              key :title, 'Array'
+              items do
+                key :title, 'DirectoryListing'
+                key :'$ref', :DirectoryListing
+              end
+            end
+          end
+          response 401 do
+            key :description, 'User is not authenticated'
+          end
+          response 403 do
+            key :description, 'User is not authorized to edit Study'
+          end
+          response 404 do
+            key :description, 'Study is not found'
+          end
+          response 406 do
+            key :description, 'Accept or Content-Type headers missing or misconfigured'
+          end
+        end
+      end
+
+      # GET /single_cell/api/v1/studies/:study_id/directory_listings
+      def index
+        @directory_listings = @study.directory_listings
+      end
+
+      swagger_path '/studies/{study_id}/directory_listings/{id}' do
+        operation :get do
+          key :tags, [
+              'DirectoryListings'
+          ]
+          key :summary, 'Find a DirectoryListing'
+          key :description, 'Finds a single Study'
+          key :operationId, 'study_directory_listing_path'
+          parameter do
+            key :name, :study_id
+            key :in, :path
+            key :description, 'ID of Study'
+            key :required, true
+            key :type, :string
+          end
+          parameter do
+            key :name, :id
+            key :in, :path
+            key :description, 'ID of DirectoryListing to fetch'
+            key :required, true
+            key :type, :string
+          end
+          response 200 do
+            key :description, 'DirectoryListing object'
+            schema do
+              key :title, 'DirectoryListing'
+              key :'$ref', :DirectoryListing
+            end
+          end
+          response 401 do
```

Comment:
How feasible would it be to have a method like "handle_4xx_errors(response)" to reduce duplicate code?  Most endpoints seem to have very similar code to handle 401-406 errors.

Not blocking; just to consider as a possible future improvement for maintainability.

Reply:
As far as I know, Swagger needs each response detailed individually.  However, there may be a way to extract these into a common library to reduce duplication.